.github/actions/export-ocm-fragments/action.yaml

```diff
@@ -161,34 +170,82 @@ runs:
 
         pprint.pprint(ocm_artefacts)
 
-        ocm_artefacts_yaml_bytes = yaml.safe_dump(ocm_artefacts).encode('utf-8')
-        ocm_artefacts_digest = hashlib.sha1(ocm_artefacts_yaml_bytes).hexdigest()
-        ocm_artefacts_fname = f'{ocm_artefacts_digest}.ocm-artefacts'
-        if ctx := '${{ inputs.ctx }}':
-          ocm_artefacts_fname = f'{ctx}-{ocm_artefacts_fname}'
-
         ocm_tar = tarfile.open(
-          name=tempfile.NamedTemporaryFile(delete=False).name,
+          name=tempfile.NamedTemporaryFile(
+            delete=False,
+            dir='.', # do not use /tmp to avoid unnecessary copy after finalised
+          ).name,
           mode='w:gz',
         )
 
-        ocm_artefacts_info = tarfile.TarInfo(name=ocm_artefacts_fname)
-        ocm_artefacts_info.size = len(ocm_artefacts_yaml_bytes)
-        buf = io.BytesIO(ocm_artefacts_yaml_bytes)
-        ocm_tar.addfile(ocm_artefacts_info, buf)
-
+        orig_name_to_digest_names = dict()
         # add blobs
         if os.path.isdir(blobs_dir := '${{ inputs.blobs-directory }}'):
           for fname in os.listdir(blobs_dir):
             fpath = os.path.join(blobs_dir, fname)
             if not os.path.isfile(fpath):
               continue
 
+            fhash = hashlib.sha256()
+            with open(fpath, 'rb') as f:
+              while chunk := f.read(4096):
+                fhash.update(chunk)
+
+            # symlink, unless file already is named according to its content-hash-digest
+            fhashname = f'sha256:{fhash.hexdigest()}'
+            if fhash != fhashname:
```

Comment:
I am wondering why this check is necessary.
How can these values become equal if `fhashname` is derived from `fhash`?

Reply:
good catch. I actually wanted to compare fname w/ fhashname!
as to your question: this check I added, as the caller (of this action) might already name the blobfile accordingly (in which case no symlinking is required (or possible..))